tests/trigger.spec.ts

```diff
@@ -364,4 +364,33 @@ describe('trigger', () => {
 
     expect(wrapper.emitted().enter).toHaveLength(1)
   })
+
+  // https://github.com/vuejs/test-utils/issues/1854
+  it('dispatches events even with fakeTimers', async () => {
+    const handlerSpy = vi.fn()
+
+    const Component = defineComponent({
+      setup() {
+        return { handlerSpy }
+      },
+      template: `
+      <div>
+        <a @click="handlerSpy">
+          <span @click="() => {}">Ok</span>
+        </a>
+      </div>
+      `
+    })
+
+    vi.useFakeTimers()
+    const wrapper = mount(Component)
+
+    expect(handlerSpy).not.toHaveBeenCalled()
+
+    wrapper.setProps({ label: 'foo' })
+
```

Comment:
This line with `wrapper.setProps` isn't needed, I forgot to remove it from an earlier version when I was tinkering with the repro.